pkg/util/ip/ip.go

```diff
@@ -129,23 +129,12 @@ func mergeCIDRs(cidrBlocks []*net.IPNet) []*net.IPNet {
 	return cidrBlocks
 }
 
-// Function to transform Antrea IPNet to net.IPNet
+// IPNetToNetIPNet converts Antrea IPNet to *net.IPNet.
+// Note that K8s allows non-standard CIDRs to be specified (e.g. 10.0.1.1/16, fe80::7015:efff:fe9a:146b/64). However,
+// OVS will report OFPBMC_BAD_WILDCARDS error if using them in the OpenFlow messages. The function will normalize the
+// CIDR if it's non-standard.
 func IPNetToNetIPNet(ipNet *v1beta2.IPNet) *net.IPNet {
 	ip := net.IP(ipNet.IP)
-	var bits int
-	if ip.To4() != nil {
-		bits = v4BitLen
-		return &net.IPNet{IP: ip, Mask: net.CIDRMask(int(ipNet.PrefixLength), bits)}
-	} else {
-		// Since OVS will report error when using an IPv6 CIDR in the OpenFlow messages if the bit is not zero in the
-		// address but it is zero in the mask. Here using the function in "net" library to ensure the result is valid.
-		_, ipNet, _ := net.ParseCIDR(fmt.Sprintf("%s/%d", ip.String(), ipNet.PrefixLength))
-		return ipNet
-	}
-}
-
-// NetIPNetToIPNet transforms net.IPNet to Antrea IPNet
-func NetIPNetToIPNet(ipNet *net.IPNet) *v1beta2.IPNet {
-	prefix, _ := ipNet.Mask.Size()
-	return &v1beta2.IPNet{IP: v1beta2.IPAddress(ipNet.IP), PrefixLength: int32(prefix)}
+	_, normalizedIPNet, _ := net.ParseCIDR(fmt.Sprintf("%s/%d", ip.String(), ipNet.PrefixLength))
```

Comment:
If non-standard CIDRs are not common, does it make sense to check if normalization is needed or not (if not just construct an IPNet struct as the previous code)?

Reply:
Good idea. Checking normalization is needed will need masking the IP first then comparing it with the original IP, the latter step is not necessary since we already get everything required to construct the IPNet. 
I have changed the code to mask the IP before constructing the IPNet. The performance improvement as below (the second one is optimized as you suggested):
```
BenchmarkIPNetToNetIPNet-4       2033640               583 ns/op             112 B/op          7 allocs/op
BenchmarkIPNetToNetIPNet2-4      6042346               169 ns/op              56 B/op          3 allocs/op
```

Reply:
Great!